packages/rocketchat-livechat/server/methods/returnAsInquiry.js

```diff
@@ -1,21 +1,56 @@
 Meteor.methods({
-	'livechat:returnAsInquiry'(rid) {
-		if (!Meteor.userId() || !RocketChat.authz.hasPermission(Meteor.userId(), 'view-l-room')) {
-			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'livechat:saveDepartment' });
+	'livechat:returnAsInquiry'(rid, departmentId) {
+		const room = RocketChat.models.Rooms.findOneById(rid);
+
+		if (!room) {
+			throw new Meteor.Error('error-invalid-room', 'Invalid room', { method: 'livechat:returnAsInquiry' });
+		}
+
+		//When an user goes offline, for example, their user is no longer available
+		//In this case, an error is raised, so to avoid this the room user is instantiated
+		let user = Meteor.user();
+		if (!user) {
+			user = RocketChat.models.Users.findOne(room.servedBy._id);
```

Comment:
you cannot get the room agent at this point because on lines bellow you're validating if the user has permission..

on the validation bellow you **must** validate the logged in user who is performing the action.